app/services/hours_builder.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+class HoursBuilder
+  def self.build(id:)
+    new(id: id).build
+  end
+
+  attr_reader :id
+  def initialize(id:)
+    @id = id.to_i
+  end
+
+  def build
```

Comment:
Suggestion: Extracting some of these methods and making `value` return `hours_hash(unavailable_message)` (or nil I guess) if the id is zero or there's no response will tighten up this logic considerably and make it look nicer and read quicker, I think.

Reply:
Refactored.